### ISSUE TITLE
:doc: reference incorrectly cross-referenced when using singleconfluence builder

### ISSUE THREAD

**Issue by tsvi** [sphinx-contrib/confluencebuilder#448]
I have a multiple-page document, but I need to publish the documentation as a single page.
When using the `:doc:` directive for referencing, the singleconfluence builder points the anchor to the filename instead of the section name.

See the following demo:

Source file tree:
 - index.rst
 - overview.rst - has a pointer to example.rst
   ```
   See the section on :doc:`Writing an XML <example>`
   ```
 - example.rst

Output conf file:
```
see the section on <ac:link ac:anchor="example">
<ac:link-body>Writing an XML</ac:link-body>
</ac:link>
```

When looking at the link generated by confluence:
- redacted/display/Area/PageName#PageNameTitle-**example**
But the anchor points to:
- redacted/pages/viewpage.action?pageId=1789830485#PageNameTitle-**Writinganxml**

**Comment by jdknight** [comment]
@tsvi, thanks for reporting this issue. A change (#454) has been added into the main branch that should address this issue and should be made available next stable release (v1.5).

If you still experience this issue after this change, please feel free to let know.

**Comment by jdknight** [comment]
v1.5 is now available on PyPI -- marking as closed.
